charts/apisix-ingress-controller/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) .Values.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "apisix-ingress-controller.fullname" . }}
+  {{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- end }}
+spec:
+  endpoints:
+  - targetPort: http
```

Comment:
Note in newer APISIX releases, APISIX using a separate port to expose Prometheus. So the `targetPort` might not be `http`, also, the Service definition should also be adjusted.

Reply:
This port is only used to expose the metrics of apisix-ingress-controller itself, not APISIX

Reply:
Got it.